### ISSUE TITLE
Font system bug

### ISSUE THREAD

**Issue by miso-nikomi-nasu** [CleverRaven/Cataclysm-DDA#33783]
Hello, nice to meet you.
I am a Japanese user and cannot speak English. This sentence uses google translation.
I think there are some unsightly points. Even if you get a response, the response may or may not be slow.

# Describe the bug

From the stable version to the latest version, there is a bug that continues to be left around the font.
Even if the font setting is changed, it will not be reflected, and the size will be changed without permission.
This bug has been verified a bit by the CataClysm: DDA community in Japan
Since there are people who have problems and who do not, problems may not be confirmed in your environment.

# Steps To Reproduce

Launch the game.

Even if the font setting is set to a different one, the problem will not change.
This setting is commonly used.

    "fontblending": false,
    "fontwidth": 8,
    "fontheight": 16,
    "fontsize": 15,
    "typeface": "MS Mincho",
    "map_fontwidth": 8,
    "map_fontheight": 17,
    "map_fontsize": 16,
    "map_typeface": "MS Mincho",
    "overmap_fontwidth": 8,
    "overmap_fontheight": 17,
    "overmap_fontsize": 16,
    "overmap_typeface": "MS Mincho"

# Expected behavior

When there was no change in size automatically
There was no problem with the font system at the time of 0.C7353
It was in the time of 0.C8069. From the time of the font system [automatic size change], it started to become difficult to use (it worked normally just because it was difficult to use).
By the time of 0.D8574, it is difficult to play due to a bug.
After that. It seems that bugs have not been discovered until 0.D9579.

Tile, console. 64bit, 32bit.
0.D has a bug in all distributed versions.

# Screenshots

0.C-7353 (Normal screen)
https://i.imgur.com/lwg61qH.png
0.C-8069 (Normal screen)
https://i.imgur.com/3m1wVPG.png
0.D-8574 (defect)
https://i.imgur.com/BrNJu5W.png
0.D-9579 (Bad)
https://i.imgur.com/JIVayGd.png
https://i.imgur.com/SygtN7k.png

# Versions and configuration

OS windows7

Game Version:
cataclysmdda-0.D-8574-Win-Curses
cataclysmdda-0.D-8574-Win-Tiles
cataclysmdda-0.D-8574-Win64-Tiles
cataclysmdda-0.D-8574-Win64-Curses

0.D-4155-g2afc018 [64-bit]
0.D-6557-g27b909c [64-bit]
0.D-7228-g146de60 [64-bit]

**Comment by lispcoc** [comment]
In screenshot from 0.D-9579 and 0.D-8574, Japanese characters are indeed "MS Mincho" font. But ascii characters are not. So I guess you do not use Windows-standard font but self-modified font.
I will get more hint probably if you upload "fontlist.txt" at all versions.

**Comment by miso-nikomi-nasu** [comment]
The other day, I understood that it was solved.

Other than the old version does not work properly and the cause remains unknown.
I haven't understood anything for nearly a year.
It was solved today without knowing anything.

Because it was garbled for some reason when I started the other day,
If you delete txt in the font list and re-output it, it will work correctly.
(I tried to delete font list txt several times in the past.)

Why there was no problem with the old version,
Why was the bug in the new version?
The reason why there was a defect for a year
I don't know what caused the problem and what caused the problem ... I'm sorry.
